chain/chain/src/flat_storage_resharder.rs

```diff
@@ -613,15 +606,37 @@ impl FlatStorageResharder {
         }
     }
 
+    /// checks whether there's a snapshot in progress, and calls split_done() if we've reached a height close to the
+    /// desired snapshot height (the state snapshot code will apply the other deltas). Returns true if we've already
+    /// applied all deltas up to the desired snapshot height, and should no longer continue to give the state snapshot
+    /// code a chance to finish first.
+    fn coordinate_snapshot(&self, height: BlockHeight, chain_store: &ChainStore) -> bool {
+        let manager = self.runtime.get_flat_storage_manager();
+        let Some(snapshot_hash) = manager.snapshot_wanted() else {
+            manager.split_done();
+            return false;
+        };
+        let Ok(snapshot_header) = chain_store.get_block_header(&snapshot_hash) else {
+            // This probably shouldn't happen, but we'll just proceed if it does, and snapshots probably won't work.
+            tracing::error!(target: "resharding", %snapshot_hash, "Could not find header for snapshot hash. Proceeding with resharding catchup");
+            manager.split_done();
+            return false;
+        };
+        let snapshot_height = snapshot_header.height();
+        if height + 10 >= snapshot_height {
```

Comment:
Is `+ 10` enough?

Reply:
I think any nonnegative int should be correct here. I moved this part to state_snapshot_actor.rs after implementing your other suggestion, but it's basically the same logic. Here the value `10` means we start trying to create the state snapshot once the resharding catchup code has made it to within 10 blocks of the desired snapshot height. We could just get rid of this whole thing and do it right away once the state is set to `Resharding(CatchingUp)`, but it would duplicate a lot of work if there are a lot of deltas to apply since now both the resharding catchup code and the snapshot code are going to be applying the same deltas on 2 different rocksdbs. So feels like a good idea to just wait until it's almost done to proceed

Reply:
I understand, that makes sense!